rust/operator-binary/src/backend/tls.rs

```diff
@@ -307,26 +311,34 @@ impl SecretBackend for TlsGenerate {
             })
             .context(BuildCertificateSnafu { tpe: CertType::Pod })?
             .build();
-        Ok([
-            (
-                "ca.crt".into(),
-                self.ca_cert
-                    .to_pem()
-                    .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
-            ),
-            (
-                "tls.crt".into(),
-                pod_cert
-                    .to_pem()
-                    .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
-            ),
-            (
-                "tls.key".into(),
-                pod_key
-                    .private_key_to_pem_pkcs8()
-                    .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
-            ),
-        ]
-        .into())
+        Ok(SecretContents::new(
+            [
+                (
+                    "ca.crt".into(),
+                    self.ca_cert
+                        .to_pem()
+                        .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
+                ),
+                (
+                    "tls.crt".into(),
+                    pod_cert
+                        .to_pem()
+                        .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
+                ),
+                (
+                    "tls.key".into(),
+                    pod_key
+                        .private_key_to_pem_pkcs8()
+                        .context(SerializeCertificateSnafu { tpe: CertType::Pod })?,
+                ),
+            ]
+            .into(),
+        )
+        .expires_after(time_datetime_to_chrono(expire_pod_after)))
     }
 }
+
+fn time_datetime_to_chrono(dt: time::OffsetDateTime) -> chrono::DateTime<FixedOffset> {
```

Comment:
Would you mind adding a test for this function? I personally find the conversion not straight forward, on small example conversion as test-case would help